src/Entity/Goal.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Acquia\LiftClient\Entity;
+
+use Acquia\LiftClient\Exception\LiftSdkException;
+use Acquia\LiftClient\Utility\Utility;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\CustomNormalizer;
+
+class Goal extends \ArrayObject
```

Comment:
Hmm, I could do that indeed. I'll move the Trait getEntityValue to that class and remove the Trait. I won't do it in this PR, as I have a couple of other PRs still to go but I'll incorporate it there